### PR TITLE
Fix #4862 - ORPHA inheritance patterns hidden

### LINKED ISSUE #4862: Do not use ORPHA inheritance pattern on genes
**Describe the bug**
It appears the mapping between gene and disease is not ideal for ORPHA disorders. We get all possible inheritance models for higher level ORPHA disorders connected to the genes. While correct at the disorder level, it can be rather misleading when presented on the gene or variant level:

**To Reproduce**
1. Go to a case
2. Pin a variant in a gene with an ORPHA disorder
3. Check the report

**Expected behavior**
Let's check if this is a parsing issue, or if it is unavoidable with the higher level terms. If the latter, lets hide the inheritance patterns at least for ORPHA that are connected to more than one gene.

**Screenshots**

Screenshot 2024-09-17 at 10 24 30

### PR DESCRIPTION
**Pull request by dnil**
This PR adds a functionality or fixes a bug.

Hide ORPHA inheritance patterns on general report, as they can be misleading. Compare note on variant view. 
Also added colored badges for the OMIM inheritance models for consistency with the variant page.

Before: 
Screenshot 2024-09-17 at 15 59 27

After:
Screenshot 2024-09-17 at 15 56 05

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [x] tests executed by DN

### CODE REVIEW

**Comment by dnil** [comment]
Tested on stage. I was kind of expecting to have to split the general css into a colors part and a more formatting oriented one and only have the colors go to the report, but looks like it works, doesn't it? See if you see anything, and I'll give it another go as well with more rested eyes later.

File: scout/server/templates/report_base.html [hunk 1]
```diff
@@ -6,6 +6,7 @@
 	{% block css %}
   <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.1.3/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-1BmE4kWBq78iYhFldvKuhfTAU6auU8tT94WrHftjDbrCEXSU1oBoqyl2QvZ6jIW3" crossorigin="anonymous">
   <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.15.4/css/all.min.css" integrity="sha512-1ycn6IcaQQ40/MKBW2W4Rhis/DbILU74C1vSrLJxCq57o941Ym01SwNsOMqvEBFlcgUa6xLiPY/NS5R+E6ztJQ==" crossorigin="anonymous" referrerpolicy="no-referrer" />
+  <link rel="stylesheet" href="{{ url_for('static', filename='bs_styles.css') }}">
```
Comment (northwestwitch) [inline scout/server/templates/report_base.html:9]: 🥇

File: scout/server/blueprints/cases/templates/cases/case_report.html [hunk 2]
```diff
@@ -712,7 +713,9 @@ <h4>Scout case analysis report</h4> - created on:&nbsp;<strong>{{report_created_
                 <td>
                   <ul class="p-0" style="list-style-type: none;">
                   {% for disease_term in gene.disease_terms %}
-                    <li class="d-flex align-items-baseline"><span class="badge bg-secondary m-1">{{ disease_term._id}}</span>&nbsp;<span>{{ disease_term.description }}&nbsp;{{ disease_term.inheritance }}</span></li>
+                    <li class="d-flex align-items-baseline"><span class="badge bg-secondary m-1">{{ disease_term._id}}</span>&nbsp;<span>{{ disease_term.description }}&nbsp;{% if disease_term.source != 'ORPHA' and disease_term.inheritance %}
+                      {% for model in disease_term.inheritance %} {{ inheritance_badge(model,inherit_palette) }}{% endfor %}
```
Comment (northwestwitch) [inline scout/server/blueprints/cases/templates/cases/case_report.html:717]: Wound't be confusing that variant page shows different values for in the inheritance?

**Review by northwestwitch — COMMENTED** [review]
This changes the code so it will show inheritance only for OMIM terms. 

But check for instance at this variant --> https://scout-stage.scilifelab.se/cust002/17083/161ee740fbbc95136ea4c0ff8cc3a916

I think it is safe enough to show that also the ORPHA inhteritance is AR. We could modify both variant page and the report so that whenever OMIM and ORPHA inheritance matches then it's shown in the ORPHA as well, and then we can show other models that appear on ORPHA but not on OMIM apart. Something like this:

```
OMIM: AR

ORPHA: AR (AD, XR..)
```

**Comment by northwestwitch** [comment]
Good that it's a thing that can be fixed without modifying the disease terms code by the way! 😆

File: scout/server/blueprints/cases/templates/cases/case_report.html (hunk 2 above)
Comment (dnil) [reply scout/server/blueprints/cases/templates/cases/case_report.html:717]: We have modified the variant page to accomodate this, adding a warning. Also, the more clear cut "Gene models" card is nearby. The variant page is used by very or at least somewhat experienced genetics and genomics persons, whereas the reports are often read also by doctors in training, who in particular provide medical input. In the not-yet-specialist case could be confused by an authoritative looking inheritance model, and cannot always be taken to understand the distinction between different database.
Screenshot 2024-09-18 at 09 55 57

Comment (northwestwitch) [reply scout/server/blueprints/cases/templates/cases/case_report.html:717]: Yes I remember this problem and our solution.

**Review by northwestwitch — APPROVED** [review]
I think this solution will be OK for our users. Consider adding a popover in the report page explaining that ORPHA inheritance is not accounted for Or perhaps modify the existing popover on the variant page adding that the ORPHA models will be not shown on the case report?

**Comment by dnil** [comment]
Right, it's a bit of a corollary to https://github.com/Clinical-Genomics/scout/pull/4443. I didn't quite think about the non-specialists reading the report then and was happy with the OMIM and ORPHA being on different lines. Updating the tooltip is a nice suggestion - will do!

File: scout/server/blueprints/variant/templates/variant/gene_disease_relations.html [hunk 3]
```diff
@@ -50,7 +50,7 @@
               {% if gene.common and gene.disease_terms %}
                 {% for disease_term in gene.disease_terms %}
                   {% if disease_term.source == 'ORPHA' %}
-                    <tr data-bs-toggle="tooltip" title="Some ORPHA disorders are phenotypic umbrella terms for multiple genetic entities. The inheritance models are in this case a set derived from all those entities, not necessarily the inheritance mode known for this gene.">
+                    <tr data-bs-toggle="tooltip" title="Some ORPHA disorders are phenotypic umbrella terms for multiple genetic entities. The inheritance models are in this case a set derived from all those entities, not necessarily modes of inheritance known for this gene. ORPHA inheritance modes will not be shown on the general case report.">
```
Comment (northwestwitch) [inline scout/server/blueprints/variant/templates/variant/gene_disease_relations.html:53]: 👍🏻

**Pull request merged**